### PR TITLE
feat(bedrock): static credentials parity via .baml options (#254 item 2)

### PR DESCRIPTION
**Pull request by invakid404**
## Summary

Implements `.baml`-declared static AWS credentials for `aws-bedrock`. Rides on the introspect-side options layer + `BedrockAuthOptions.Credentials` slot from #262. Closes `#254` item 2; umbrella tracker stays open for reasoning sig/redacted, tool-use streaming, and usage metadata.

Operators can now do:

```baml
client<llm> Bedrock {
  provider aws-bedrock
  options {
    access_key_id     env.AWS_ACCESS_KEY_ID
    secret_access_key env.AWS_SECRET_ACCESS_KEY
    session_token     env.AWS_SESSION_TOKEN  // optional
    profile           env.AWS_PROFILE
    region            "us-east-1"
  }
}
```

(literal values also supported — see security note).

## Resolution precedence

Matches BAML's own at `engine/baml-runtime/.../aws_client.rs:587-650`:
1. **Static** (`access_key_id` + `secret_access_key` + optional `session_token`) → `credentials.NewStaticCredentialsProvider`.
2. **Profile** → `config.LoadDefaultConfig(ctx, config.WithSharedConfigProfile(name))`.
3. **Default chain** — unchanged from #262.

Static wins over profile when both configured. Default chain unchanged for clients without credential options.

## What's included

- `cmd/introspect` parser extension capturing `access_key_id`, `secret_access_key`, `session_token`, `profile` with literal-vs-`env.X` provenance.
- New nested `BedrockCredentialOptions` struct on `BedrockClientOptions`.
- `bamlutils/llmhttp.ResolveBedrockCredentials(ctx, clientName, sel)` applying the precedence above.
- `AttachBedrockAuthForClient` migrated to an options struct (`BedrockClientAuthOptions`) so future extensions don't keep growing the signature.
- Test matrix: parser/emission (8+ cases), resolver (7+ cases incl. invalid partial states), signing-end-to-end (static-vs-default sig divergence), full E2E call/stream through `httptest.NewServer`, security-pin test asserting errors never leak credential values.

## Security note

Literal `.baml` credentials are BAML-parity behavior — supported here for that reason. Literal values land in generated `introspected.go` and the compiled binary; the new `BedrockCredentialOptions` doc-comment flags this and recommends `env.X` as the safer form. No error path echoes credential values; logging hygiene preserved.

## What's NOT included (deferred to follow-ups under #254)

- Reasoning `signature` / `redactedContent` exposure (item 3).
- Tool-use streaming deltas (item 4).
- Usage metadata propagation (item 5).
- `AWS_DEFAULT_REGION` region fallback — flagged by scoping but separate scope from item 2; #262's region contract preserved as-is.

## Verification

- `go build ./...` / `go vet ./...` clean.
- `verify-framework-adapter`: 9/9. `verify-adapter-pins`: 4/4.
- `gofmt -l` clean on touched files.
- `go run ./cmd/embed && git status --porcelain` clean.

Refs #254.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Bedrock authentication now accepts a single client auth options payload that includes endpoint, region, and per-field credential selectors (access key, secret, session token, profile); empty options fall back to the default AWS credential chain.

* **Tests**
  * Added extensive unit and end-to-end tests covering static credentials, profile loading, precedence, signing, and validation edge cases.

* **Documentation**
  * Introspection/config output now exposes credential selector fields and an IsSet() declaration-vs-resolve indicator; notes on inline-secret handling added.

[Review Change Stack](https://app.coderabbit.ai/change-stack/invakid404/baml-rest/pull/263)

### CODE REVIEW

File: bamlutils/llmhttp/awssigv4_e2e_test.go [hunk 1]
```diff
@@ -237,6 +245,121 @@ func TestEndpointOverride_E2E_RegionFallback(t *testing.T) {
...
+		t.Fatalf("ExecuteAWSStream: %v", err)
+	}
+	defer resp.Close()
+	if _, err := resp.Events.Next(); err != nil && err != io.EOF {
+		t.Fatalf("Events.Next: %v", err)
+	}
+
+	if !strings.Contains(gotAuth, "Credential=STATIC_TEST_ACCESS_KEY/20260511/us-east-1/bedrock/aws4_request") {
+		t.Errorf("Authorization must carry static access key in credential scope on the streaming path; got: %s", gotAuth)
+	}
```
[the remark this answers is not in the document]
Comment (invakid404) [reply bamlutils/llmhttp/awssigv4_e2e_test.go:386]: @coderabbitai Good catch — fixed in e19221274.

Both `TestStaticCreds_E2E_NonStreaming` and `TestStaticCreds_E2E_Streaming` now record `X-Amz-Security-Token` from the request headers and assert it equals `"STATIC_TEST_SESSION_TOKEN"` after dispatch. The non-streaming variant didn't configure a session token at all previously; added one for call/stream parity per Codex's symmetry note during verification.

This proves the full path end-to-end: resolver-created static provider (with session token) → SigV4 copy-back into `req.Headers` (already pinned at unit level by `TestSignRequest_SessionToken`) → `Execute` / `ExecuteAWSStream` → wire headers. A regression that drops the session token anywhere in that chain will now fail at the E2E layer too.

While in there, also fixed an unrelated sign-off blocker: `BedrockCredentialSelector.*Present` was using `BedrockOptionValue.Resolve()`'s `ok` flag, which conflates "declared but env unset" with "never declared". Replaced with a new `BedrockOptionValue.IsSet()` driven by `Provenance != ""`. Declared env refs that resolve unset now error explicitly instead of silently falling back to the default AWS credential chain. Regression pinned by `TestResolveBedrockCredentials_EnvRefAllUnset_Errors` + `_ProfileEnvUnset_Errors`.

File: adapters/common/codegen/codegen_buildrequest.go [hunk 2]
```diff
@@ -65,20 +88,33 @@ func emitBedrockAuthDispatchFor(methodName string) func(*jen.Group) {
...
 				jen.Id("ctx"),
 				jen.Id("req"),
-				jen.Id("bedrockEndpointURL"),
-				jen.Id("bedrockRegion"),
+				jen.Qual(common.LLMHTTPPkg, "BedrockClientAuthOptions").Values(jen.Dict{
+					jen.Id("ClientName"):  jen.Id("selectedClient"),
+					jen.Id("EndpointURL"): jen.Id("bedrockEndpointURL"),
+					jen.Id("Region"):      jen.Id("bedrockRegion"),
+					jen.Id("Credentials"): jen.Id("bedrockCreds"),
+				}),
```
[the remark this answers is not in the document]
Comment (invakid404) [reply adapters/common/codegen/codegen_buildrequest.go:135]: @coderabbitai Good catch — fixed in 6a6584910.

You're right that this was the same root cause as the credentials bug from the prior commit, just on a different pair of fields. Same fix shape applied symmetrically:

- `BedrockClientAuthOptions` gains `EndpointURLPresent` and `RegionPresent` flags.
- Codegen drives them from `BedrockOptionValue.IsSet()` (declaration), with values still from `Resolve()` (resolution). Mirrors the credentials pattern at the same call site.
- `AttachBedrockAuthForClient` now errors with a client-scoped message when `Present=true && value==""` for either field — same shape as the credentials resolver:
  - `aws-bedrock: invalid endpoint_url for client "X": endpoint_url resolved to empty`
  - `aws-bedrock: invalid region for client "X": region resolved to empty`

**Most concrete failure mode this catches**: an operator declares `endpoint_url env.MY_PROXY` (intending to route to a custom proxy / LocalStack / VPC endpoint) and forgets to set `MY_PROXY` at runtime. Before this fix: silently sends to real AWS Bedrock. After: explicit error pointing at the broken declaration.

**Compatibility preserved**: direct callers passing `BedrockClientAuthOptions{EndpointURL: "http://h"}` without the explicit `EndpointURLPresent` flag still work — the helper treats a non-empty raw string as present too. Pinned by `TestAttachBedrockAuthForClient_NonEmptyRawStringTreatedAsPresent`.

**AWS_REGION env fallback** is preserved as-is in `AttachBedrockAuthWithOptions`. It only kicks in when region was NOT declared in `.baml`. When region IS declared as `env.X` and X is unset, the helper-level error fires before that fallback could be reached — `TestAttachBedrockAuthForClient_RegionEnvUnset_Errors` sets `AWS_REGION=ap-southeast-2` to prove this ordering.

Regression pins added: `_EndpointURLEnvUnset_Errors`, `_RegionEnvUnset_Errors`, `_NeitherDeclared_FallsThrough`, `_NonEmptyRawStringTreatedAsPresent`, `_EndpointURLDeclaredEmptyBeforeRegionFallback`. Codegen emission pin extended in `TestEmitBedrockAuthDispatchFor_Shape`.
